adsft/extraction.py

```diff
@@ -393,14 +391,11 @@ def open_xml(self):
             with open(self.file_input, 'rb') as fp:
                 raw_xml = fp.read()
 
-            # use soupparser to properly encode file contents
-            #  it could be utf-8, iso-8859, etc.
-            parsed_content = soupparser.fromstring(raw_xml)
-            # convert to string for ease of clean-up, convert html and LaTeX entities
-            raw_xml = tostring(parsed_content)
-            raw_xml = re.sub('(<!-- body|endbody -->)', '', raw_xml)
+            encoding = UnicodeDammit(raw_xml).originalEncoding
+
+            raw_xml = raw_xml.decode(encoding, "ignore")
+
             raw_xml = edef.convertentities(raw_xml)
-            raw_xml = re.sub('<\?CDATA.+?\?>', '', raw_xml)

```

Comment:
As the one above, do you know why this would not be necessary anymore?

Reply:
I'm removing tex-math tags which always wrap around these ?CDATA tags.

Reply:
Let’s try the `drop_tag` and see if that works, even if it is unnecessary for your sample maybe it is necessary for other documents in our database... who knows.

Reply:
BeautifulSoup is parsing ?CDATA tags as processing instructions, and this results in a that CDATA tag never being added to the xml tree. This means we can't use any of the functions like drop_tree, drop_tag etc. to remove it. We could go back to using regex but this isn't full proof because a CDATA tag can span multiple lines. The only way to be 100% sure that CDATA tags are removed is to modify BeautifulSoup.

Reply:
Sorry I'm late to the conversation.  Is this removing CDATA just for the fulltext searching?  If so, great, but if it's text that we're actually going to serve, Carolyn asked me to find a way to retain it.  The CDATA and math tags occasionally appear in JATS abstracts too, and I'm trying to find a way not to discard them.

Reply:
Yes it's just for fulltext searching, all I extract in this pipeline is the body, acknowledgements and appendix. 

If your CDATA is in the form <?CDATA some data ?> then you can use the xpath '//processing-instruction()' to access that data. You are supposed to be able to pass a parameter like //processing-instruction('CDATA') but I found this did not work in my case. I just updated this PR to remove all processing instructions.